planner/core/planbuilder.go

```diff
@@ -1755,18 +1781,45 @@ func (b *PlanBuilder) buildAnalyzeIndex(as *ast.AnalyzeTableStmt, opts map[ast.A
 	if err != nil {
 		return nil, err
 	}
+	statsHandle := domain.GetDomain(b.ctx).StatsHandle()
+	if statsHandle == nil {
+		return nil, errors.Errorf("statistics hasn't been initialized, please try again later")
+	}
+	versionIsSame, err := statsHandle.CheckAnalyzeVersion(tblInfo, physicalIDs, &version)
+	if err != nil {
+		return nil, err
+	}
+	if !versionIsSame {
+		if b.ctx.GetSessionVars().EnableFastAnalyze {
+			return nil, errors.Errorf("Fast analyze hasn't been GA and is not available for new statistics.")
+		}
+		b.ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("The analyze version from the session is not compatible with the existing statistics of the table. Use the old one instead"))
+	}
```

Comment:
I think we're assuming the original stats ver is 1 and the user set it to 2 here. But it's also possible that the original stats ver is 2 and the user set it back to 1.
In that case, we can do fast analyze, and also the warning should not be 'the old one'.